platform/eureka/client.go

```diff
@@ -0,0 +1,97 @@
+package eureka
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type application struct {
+	Name      string      `json:"name"`
+	Instances []*instance `json:"instance"`
+}
+
+type instance struct {
+	Hostname   string   `json:"hostName"`
+	App        string   `json:"app"`
+	IPAddress  string   `json:"ipAddr"`
+	Status     string   `json:"status"`
+	Port       *port    `json:"port,omitempty"`
+	SecurePort *port    `json:"securePort,omitempty"`
+	Metadata   metadata `json:"metadata,omitempty"`
+}
+
+type port struct {
+	Port    int  `json:"$"`
+	Enabled bool `json:"@enabled,string"`
+}
+
+type metadata map[string]string
+
+// Client for Eureka
+type Client interface {
+	// Applications registered on the Eureka server
+	Applications() ([]*application, error)
+}
+
+// Minimal client for Eureka server's REST APIs.
+// TODO: support multiple Eureka servers
+// TODO: caching
+type client struct {
+	client http.Client
+	url    string
+}
+
+// NewClient instantiates a new Eureka client
+func NewClient(url string) Client {
+	return &client{
+		client: http.Client{Timeout: 30 * time.Second},
+		url:    url,
+	}
+}
+
+const statusUp = "UP"
+
+const (
+	basePath = "/eureka/v2"
+	appsPath = basePath + "/apps"
+)
+
+type getApplications struct {
+	Applications applications `json:"applications"`
+}
+
+type applications struct {
+	Applications []*application `json:"application"`
```

Comment:
No, Eureka is weird and the field name is `application`, even though it is an array of applications structures. In terms of the JSON field matching the struct field name, that is a style preference. In other places in the code, we don't strictly adhere to the JSON field name because some of them are misleading or use a different abbreviation style than the rest of the code.